cdm/misc/src/main/java/ucar/nc2/geotiff/GeotiffWriter.java

```diff
@@ -213,8 +215,16 @@ private void writeMetadata(boolean greyScale, double xStart, double yStart, doub
       geotiff.addTag(new IFDEntry(Tag.XResolution, FieldType.RATIONAL).setValue(1, 1));
       geotiff.addTag(new IFDEntry(Tag.YResolution, FieldType.RATIONAL).setValue(1, 1));
       geotiff.addTag(new IFDEntry(Tag.ResolutionUnit, FieldType.SHORT).setValue(1));
-
+      if (colorTable != null && colorTable.length > 0) {
+        geotiff.addTag(new IFDEntry(Tag.PhotometricInterpretation, FieldType.SHORT).setValue(3));
+        geotiff.addTag(new IFDEntry(Tag.ColorMap, FieldType.SHORT, colorTable.length).setValue(colorTable));
+      } else {
+        // black is zero (value used in GeotiffWriter)
+        geotiff.addTag(new IFDEntry(Tag.PhotometricInterpretation, FieldType.SHORT).setValue(1));
+      }
```

Comment:
Probably need to do a bit of thinking about this method's call signature, particularly the `greyscale` argument. Need to consider how orthogonal these features are.

Reply:
Yeah, this isn't working out too great conceptually (and it is making patching difficult in tds). Since this is a private method, can I take that to mean I can change up the call signature a bit? Basically, I'm thinking of adding a new argument `dtype` that can specify what `DataType` the output should be (or should I use `FieldType`?). If it is `null`, then it means "auto" and it'll match the dtype of the given data.

For backwards compatibility, I could ignore `dtype` if greyscale is true.

Thoughts? It'll require a new writeData() for integer type (there is currently one for byte and float). Probably wouldn't cover everything geotiff can handle, but should be pretty decent, right?

Reply:
yes,  you can change the signature for this method since it is private.

Reply:
Great! I kind of want to split this function up a bit so that the "greyscale" stuff is out of here and can be handled separately. It would make this function and my proposed `dtype` approach much more sane and reasonable. We can then relegate the `greyscale` handling as a special-case that doesn't care about any of the dtype information.